pytube/extract.py

```diff
@@ -171,7 +172,8 @@ def channel_name(url: str) -> str:
         r"(?:\/(c)\/([%\d\w_\-]+)(\/.*)?)",
         r"(?:\/(channel)\/([%\w\d_\-]+)(\/.*)?)",
         r"(?:\/(u)\/([%\d\w_\-]+)(\/.*)?)",
-        r"(?:\/(user)\/([%\w\d_\-]+)(\/.*)?)"
+        r"(?:\/(user)\/([%\w\d_\-]+)(\/.*)?)",
+        r"(?:\/(\@)([%\d\w_\-]+)(\/.*)?)"
```

Comment:
I think that with this regex `r"(?:\/(@[%\d\w_\-]+)()(\/.*)?)` it's possible to avoid @ checks below